File: Go/Sum of Two Numbers/solution_3.go
package main

import "sort"

// O(nlog(n)) time | O(1) space
func TwoNumberSum(array []int, target int) []int {

	sort.Ints(array)
	first_ptr := 0
	second_ptr := len(array) - 1

	for first_ptr < second_ptr {

		currentSum := array[first_ptr] + array[second_ptr]

		if currentSum == target {
			return []int{array[first_ptr], array[second_ptr]}
		} else if currentSum < target {
			first_ptr += 1
		} else {
			second_ptr -= 1
		}

	}

	return []int{}
}

File: Python/Sum of Two Numbers/solution_3.py
def twoNumberSum(array, targetSum):

    array.sort()
    first = 0
    second = len(array) - 1

    while first < second:

        currentSum = array[first] + array[second]

        if currentSum == targetSum:
            return [array[first], array[second]]
        elif currentSum < targetSum:
            first += 1
        elif currentSum > targetSum:
            second -= 1

    return []
        
        
